### PR TITLE
Test/jpa repository

### LINKED ISSUE #38: Test JPA repositories
Test the JPA using `@DataJPATest`, inject dependencies with `@Autowired` and test the repositories. This creates an in-memory database and does not affect our real database. Also, the testdatabase is cleaned between each test to avoid dependencies between tests. 

**PollRepository**   

- [x] When persisting a poll it should be stored in the database   

- [x] When deleting a poll it should delete the votes on the poll   

- [x] Deleting a poll it does not delete the owner 

**VoteRepository**   

- [x] When persisting a vote it is added to the poll   

- [x] When persisting a vote it is added to the users votes   

**VoterRespitory**      

- [x] When deleting a voter the votes are not deleted

### PR DESCRIPTION
**Pull request by JennStro**
Closes #38 

This PR is possible solution to #54 if we restrict setPolls(polls...) to be set in user. 

#### Changed   
*   Primary Key to PollRepository from String to Long in `public interface PollRepository extends JpaRepository<Poll, String> {`  to `public interface PollRepository extends JpaRepository<Poll, Long> {` because it was the wrong type.    
*   Changed from `long` to `Long` for consistancy. 

#### Added   
Tests for repositories, see #38

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -104,6 +104,18 @@
...
+            <artifactId>junit-jupiter-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
+        </dependency>
```
Comment (oddhus) [inline pom.xml]: Disse er ikke nødvendige, tror at spring-boot-starter-test dekker alt foreløpig

**Review by oddhus — DISMISSED** [review]
Veldig mange gode og dekkende tester! 🥇 Det blir lett å oppdage problemer underveis nå 👍

File: src/main/lombok/no/hvl/dat250/h2020/group5/entities/Poll.java [hunk 2]
```diff
@@ -25,6 +23,7 @@
     @SequenceGenerator(name="PollID_Sequence", initialValue = LOWEST_4_DIGIT_BASE36)
     @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "PollID_Sequence")
     @Convert(converter = AlphaNumeric2Long.class)
+    @Setter(AccessLevel.NONE)
```
Comment (oddhus) [inline src/main/lombok/no/hvl/dat250/h2020/group5/entities/Poll.java]: Jeg har noen tester som avhenger av å sette Id'en :S

I tillegg bør man visst ha setters hvis man har getter fant jeg it her: https://stackoverflow.com/questions/12369641/is-there-any-reason-to-not-generate-setters-and-getters-for-id-fields-in-jpa

Comment (JennStro) [reply src/main/lombok/no/hvl/dat250/h2020/group5/entities/Poll.java]: Aha, har endret det nå :)

File: pom.xml (hunk 1 above)
Comment (JennStro) [reply pom.xml]: Det funket fint uten de ja =)

**Review by oddhus — APPROVED** [review]
Supert 🚀 💯

**Pull request merged**